backend/dal/Services/Concrete/BuildingService.cs

```diff
@@ -193,14 +227,132 @@ public Building Update(Building building)
+                    if (updateParcel == null)
+                    {
+                        this.ThrowIfNotAllowedToUpdate(parcelBuilding.Building, _options.Project);
+
+                        var parcelBuildings = this.Context.ParcelBuildings.Where(pb => pb.ParcelId == parcelBuilding.ParcelId).ToArray();
+                        building.Parcels.Remove(parcelBuilding);
+                        this.Context.ParcelBuildings.Remove(parcelBuilding);
+
+                        continue;
+                    }
+
+                    // The parcel may have evaluations or fiscals that need to be deleted.
+                    foreach (var parcelEvaluation in parcelBuilding.Parcel.Evaluations)
+                    {
+                        // Delete the evaluations that have been removed.
+                        if (!updateParcel.Parcel.Evaluations.Any(e => (e.ParcelId == parcelEvaluation.ParcelId && e.Date == parcelEvaluation.Date && e.Key == parcelEvaluation.Key)))
+                        {
+                            this.Context.ParcelEvaluations.Remove(parcelEvaluation);
+                        }
+                    }
+                    foreach (var parcelFiscal in parcelBuilding.Parcel.Fiscals)
+                    {
+                        // Delete the fiscals that have been removed.
+                        if (!updateParcel.Parcel.Fiscals.Any(e => (e.ParcelId == parcelFiscal.ParcelId && e.FiscalYear == parcelFiscal.FiscalYear && e.Key == parcelFiscal.Key)))
+                        {
+                            this.Context.ParcelFiscals.Remove(parcelFiscal);
+                        }
+                    }
+                }
 
+                foreach (var buildingEvaluation in existingBuilding.Evaluations)
+                {
+                    // Delete the evaluations from the parcel that have been removed.
+                    if (!building.Evaluations.Any(e => (e.BuildingId == buildingEvaluation.BuildingId && e.Date == buildingEvaluation.Date && e.Key == buildingEvaluation.Key)))
+                    {
+                        this.Context.BuildingEvaluations.Remove(buildingEvaluation);
+                    }
+                }
+                foreach (var buildingFiscals in existingBuilding.Fiscals)
+                {
+                    // Delete the fiscals from the parcel that have been removed.
+                    if (!building.Fiscals.Any(e => (e.BuildingId == buildingFiscals.BuildingId && e.FiscalYear == buildingFiscals.FiscalYear && e.Key == buildingFiscals.Key)))
+                    {
+                        this.Context.BuildingFiscals.Remove(buildingFiscals);
+                    }
+                }
+            }
+            existingBuilding.LeasedLandMetadata = building.LeasedLandMetadata;
             this.Context.Buildings.Update(existingBuilding); // TODO: Must detach entity before returning it.
             this.Context.CommitTransaction();
```

Comment:
Have you tested for concurrency with `RowVersion`?

Reply:
Yes